mesonbuild/backend/ninjabackend.py

```diff
@@ -1778,19 +1793,21 @@ def generate_genlist_for_target(self, genlist, target):
                 outfilelist = outfilelist[len(generator.outputs):]
             args = self.replace_paths(target, args, override_subdir=subdir)
             cmdlist = exe_arr + self.replace_extra_args(args, genlist)
-            if generator.capture:
+            if serialize:
                 exe_data = self.serialize_executable(
                     'generator ' + cmdlist[0],
                     cmdlist[0],
                     cmdlist[1:],
                     self.environment.get_build_dir(),
-                    capture=outfiles[0]
+                    capture=outfiles[0] if generator.capture else None
                 )
                 cmd = self.environment.get_build_command() + ['--internal', 'exe', exe_data]
+             else:
```

Comment:
__[Flake8]__ [E999] IndentationError: unindent does not match any outer indentation level ([link](https://sider.review/gh/repos/19784232/pulls/5573#issue-4288104))

<sub>posted by Sider</sub>